Expose additional methods in REST Data with Panache

Sgitario opened the pull request:
With these changes, you can add additional methods to the generated resources by the REST Data with Panache extension by adding these methods to the resource interface, for example:

```java
@ResourceProperties
public interface PeopleResource extends PanacheEntityResource<Person, Long> {
    @GET
    @Path("/name/{name}")
    default List<Person> findByName(@PathParam("name") String name) {
        return Person.find("name = :name", Collections.singletonMap("name", name)).list();
    }
}
```

And this method will be exposed along with the generated methods using `http://localhost:8080/people/name/Johan`.

Related to https://groups.google.com/g/quarkus-dev/c/c2FTXsX1fNo/m/qEWtOpzlBQAJ?utm_medium=email&utm_source=footer

FroMage reviewed (approved):
LGTM. Are you sure the added links also take into account the user methods?

Sgitario committed:
Expose additional methods in REST Data with Panache

With these changes, you can add additional methods to the generated resources by the REST Data with Panache extension by adding these methods to the resource interface, for example:

```java
@ResourceProperties
public interface PeopleResource extends PanacheEntityResource<Person, Long> {
    @GET
    @Path("/name/{name}")
    default List<Person> findByName(@PathParam("name") String name) {
        return Person.find("name = :name", Collections.singletonMap("name", name)).list();
    }
}
```

And this method will be exposed along with the generated methods using `http://localhost:8080/people/name/Johan`.

Related to https://groups.google.com/g/quarkus-dev/c/c2FTXsX1fNo/m/qEWtOpzlBQAJ?utm_medium=email&utm_source=footer
